### PR TITLE
Add formatPreciseDuration

### PR DESCRIPTION
**Pull request by paulyoung**

### CODE REVIEW

**Comment by paulyoung** [comment]
I think I need to push a fix for negative durations.

**Comment by paulyoung** [comment]
We have a need to render these without intermediate zeros (e.g. "1w 1h" instead of "1w 0d 1h") so I'm going to make changes to support that in addition to what's already here.

File: src/Data/Time/PreciseDuration/Format.purs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import Data.Array as Array
+import Data.Decimal (Decimal, modulo)
+import Data.Maybe (Maybe(..))
+import Data.Monoid (guard)
+import Data.Time.PreciseDuration (PreciseDuration)
+import Data.Time.PreciseDuration as PD
+
+type Format =
+  { intermediateZeroes :: Boolean
+  }
```
Comment (paulyoung) [inline src/Data/Time/PreciseDuration/Format.purs]: This is so we could potentially have things like `leadingZeroes` and `trailingZeroes` in the future

File: src/Data/Time/PreciseDuration/Format.purs [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    zHoursXMins = hours == zero && mins > zero
+    zMinsXSecs = mins == zero && secs > zero
+
+    components :: Array (Maybe String)
+    components =
+      [ guard
+          gteWeek
+          (Just $ nsToString PD.toWeeks weeks)
+      , guard
+          (gteDay && ltWeek || gteWeek && (days > zero || format.intermediateZeroes && (zDaysXHours || zHoursXMins || zMinsXSecs)))
```
Comment (colinwahl) [inline src/Data/Time/PreciseDuration/Format.purs]: I'm not sure if there is a simpler way to do this.  This works, but it is hard to keep track of what's going on.

Comment (paulyoung) [reply src/Data/Time/PreciseDuration/Format.purs]: I feel the same way.

Comment (colinwahl) [reply src/Data/Time/PreciseDuration/Format.purs]: With the comment you added this is more clear

**Comment by paulyoung** [comment]
I agree that we should come back to this. It can probably be rewritten to use a fold.

**Pull request merged**